srml/balances/src/lib.rs

```diff
@@ -592,26 +689,33 @@ where
 	}
 }
 
-impl<T: Trait<I>, I: Instance> TransferAsset<T::AccountId> for Module<T, I> {
-	type Amount = T::Balance;
+/// Simple implementation of OnUnbalancedDecrease/OnUnbalancedIncrease that
+/// just burns and mints tokens as necessary.
+pub struct BurnAndMint<T: Trait>(rstd::marker::PhantomData<T>);
 
-	fn transfer(from: &T::AccountId, to: &T::AccountId, amount: T::Balance) -> Result {
-		Self::make_transfer(from, to, amount)
+impl<T: Trait> OnUnbalancedDecrease<T::Balance> for BurnAndMint<T> {
+	fn on_unbalanced_decrease(amount: T::Balance) -> Result {
+		<Module<T>>::decrease_total_issuance_by(amount)
```

Comment:
I don't understand how it could be useful, could you instance another implementation ?

Reply:
Yes. You might want to put it in the treasury, for example.